task/src/main/java/com/rarible/core/task/TaskRunner.kt

```
@@ -52,7 +49,7 @@ class TaskRunner(
             } else {
                 taskRepository.save(current.markCompleted()).awaitFirst()
             }
-        } catch (e: Throwable) {
```

Why did you change Throwable -> Exception? Don't it that be dangerously?

---

It's the other way round: catching Throwable is dangerous! One should never catch Throwable unless, in some cases, on the top of the stack (i.e., in Thread's Runnable). 
The thing is that Throwables that are not Exceptions are Errors, and they are not supposed to be caught in the application code. There are several risks to catch them:
1. Masking of the original problem: while processing the Error in the catch block, new Errors may arise because of the same condition. For example, if it's OutOfMemoryError, then logging it allocates memory and can cause yet another OOME. If it's a StackOverflowError, the stack at this point might be at the brink of being full, so calling any function can cause yet another SOE. And in the resulting error the original error will be lost, so we wouldn't know where it actually happened.
2. It is dangerous to proceed, because the invariants might have been broken. For the same reason as above: finally block could not have been executed fully because an error happened while it was working. For this reason, it is better to leave it to JVM to handle it.